src/assets/decompiler.ts

```diff
@@ -2,16 +2,59 @@ export interface ParsedInstruction {
   aMux: boolean;
   mbr: boolean;
   mar: boolean;
-  rdWr: boolean;
+  rdWr: ReadWrite;
   ms: boolean;
   enS: boolean;
-  cond: number;
-  alu: number;
-  sh: number;
+  cond: Condition;
+  alu: Alu;
+  sh: Shift;
   sBus: number;
   bBus: number;
   aBus: number;
   adr: number;
+  originalInstruction: string;
+}
+
+export enum InstructionParts {
+  aMux = "aMux",
+  mbr = "mbr",
+  mar = "mar",
+  rdWr = "rdWr",
+  ms = "ms",
+  enS = "enS",
+  cond = "cond",
+  alu = "alu",
+  sh = "sh",
+  sBus = "sBus",
+  bBus = "bBus",
+  aBus = "aBus",
+  adr = "adr",
+}
+
+export enum Condition {
+  Never = 0,
+  Negative = 1,
+  Zero = 2,
+  Everytime = 3
+}
+
+export enum Alu {
+  Pass = 0,
+  Plus = 1,
+  And = 2,
+  Negate = 3
+}
+
+export enum Shift {
+  Never = 0,
+  L = 1,
+  R = 2,
+  Invalid = 3
+}
+
+export enum ReadWrite {
+  Read = 1,
+  Write = 0
```

Comment:
Damn sweet, that's one well chosen use of Typescript enums!

Reply:
the read write one maybe a bit overkill, but the other ones definitely feel a lot better
ReadWrite is not really a toggle so it made more sense as a enum to me